packages/frontend-2/components/settings/workspaces/security/Discoverability.vue

```diff
@@ -0,0 +1,313 @@
+            :show-label="false"
+          />
+        </div>
+      </div>
+
+      <div v-if="isDomainDiscoverabilityEnabled" class="flex flex-col gap-2">
+        <p class="text-body-xs font-medium text-foreground">
+          When someone wants to join
+        </p>
+        <div
+          v-tippy="!isWorkspaceAdmin ? 'You must be a workspace admin' : undefined"
+          :class="!isWorkspaceAdmin ? 'opacity-70' : ''"
+          class="max-w-max"
+        >
+          <FormRadio
+            v-for="option in radioOptions"
+            :key="option.value"
+            :disabled="!isWorkspaceAdmin"
+            :label="option.title"
+            :value="option.value"
+            name="joinPolicy"
+            :checked="joinPolicy === option.value"
+            size="sm"
+            :label-classes="
+              !isWorkspaceAdmin
+                ? '!font-normal !cursor-not-allowed'
+                : '!font-normal cursor-pointer'
+            "
+            @change="handleRadioChange(option.value)"
+          />
+        </div>
+      </div>
+    </div>
+
+    <SettingsConfirmDialog
+      v-model:open="showConfirmJoinPolicyDialog"
+      title="Confirm change"
+      @confirm="handleJoinPolicyConfirm"
+      @cancel="handleJoinPolicyCancel"
+    >
+      <p class="text-body-xs text-foreground mb-2">
+        This will allow users with verified domain emails to join automatically without
+        admin approval.
+      </p>
+      <p class="text-body-xs text-foreground">Are you sure you want to enable this?</p>
+    </SettingsConfirmDialog>
+  </section>
+</template>
+
+<script setup lang="ts">
+import { Roles } from '@speckle/shared'
+import { useMutation } from '@vue/apollo-composable'
+import { graphql } from '~/lib/common/generated/gql'
+import type { SettingsWorkspacesSecurityDiscoverability_WorkspaceFragment } from '~/lib/common/generated/gql/graphql'
+import { useMixpanel } from '~/lib/core/composables/mp'
+import {
+  workspaceUpdateDiscoverabilityMutation,
+  workspaceUpdateAutoJoinMutation
+} from '~/lib/workspaces/graphql/mutations'
+
+enum JoinPolicy {
+  AdminApproval = 'admin-approval',
+  AutoJoin = 'auto-join'
+}
+
+graphql(`
+  fragment SettingsWorkspacesSecurityDiscoverability_Workspace on Workspace {
+    id
+    slug
+    role
+    domains {
+      id
+      domain
+    }
+    sso {
+      provider {
+        id
+      }
+    }
+    discoverabilityEnabled
+    discoverabilityAutoJoinEnabled
+  }
+`)
+
+const props = defineProps<{
+  workspace: SettingsWorkspacesSecurityDiscoverability_WorkspaceFragment
+}>()
+
+const mixpanel = useMixpanel()
+const { mutate: updateDiscoverability } = useMutation(
+  workspaceUpdateDiscoverabilityMutation
+)
+const { mutate: updateAutoJoin } = useMutation(workspaceUpdateAutoJoinMutation)
+const { triggerNotification } = useGlobalToast()
+
+const showConfirmJoinPolicyDialog = ref(false)
+const pendingJoinPolicy = ref<JoinPolicy>()
```

Comment:
We could use just a boolean value here, `pendingIsDiscoverabilityAutoJoinEnabled` or something, then you dont need the custom type + mapping